lib/haskell.nix

```diff
@@ -1,99 +1,85 @@
-
-     * Run a test for a specific GHC version:
-       - `nix check .#ghc<version>:<package>:test:<test>`
-
-     Inputs:
-       - The haskell.nix packages set (i.e. pkgs.haskell-nix)
-       - A haskell.nix *Project function (e.g. pkgs.haskell-nix.stackProject)
-       - Attrs with:
-         - ghcVersions: compiler versions to build with (in addition to the default one)
-         - Any other attributes that will be forwarded to the project function
-
-     Example:
-       makeFlake pkgs.haskell-nix pkgs.haskell-nix.stackProject {
-         src = ./.;
-         ghcVersions = [ "901" ];
-       }
-       =>
-       # Assuming you use `flake-utils.lib.eachSystem [ "x86_64-linux" ]`
-       $ nix flake show
-       <...>
-        ├───apps
-        │   └───x86_64-linux
-        │       ├───"ghc901:package:exe:executable": app
-        │       ├───"ghc901:package:test:test": app
-        │       ├───"package:exe:executable": app
-        │       └───"package:test:test": app
-        ├───checks
-        │   └───x86_64-linux
-        │       ├───"ghc901:package:test:test": derivation 'test-test-1.0.0-check'
-        │       └───"package:test:test": derivation 'test-test-1.0.0-check'
-        ├───devShell
-        │   └───x86_64-linux: development environment 'ghc-shell-for-package'
-        └───packages
-            └───x86_64-linux
-                ├───"ghc901:package:exe:executable": package 'package-exe-executable-1.0.0'
-                ├───"ghc901:package:lib:package": package 'package-lib-package-1.0.0'
-                ├───"ghc901:package:test:test": package 'test-test-1.0.0'
-                ├───"package:exe:executable": package 'package-exe-executable-1.0.0'
-                ├───"package:lib:package": package 'package-lib-package-1.0.0'
-                └───"package:test:test": package 'test-test-1.0.0'
-  */
-  makeFlake = haskellNix: projectF: args@{ ghcVersions, ... }:
-    let
-      args' = builtins.removeAttrs args [ "ghcVersions" ];
-
-      flakeForGhc = ghcVersion:
-        let
-          compiler-nix-name =
-            if ghcVersion == null
-            then null
-            else "ghc${ghcVersion}";
-          project = projectF (args' // { inherit compiler-nix-name; });
-          prefix =
-              if compiler-nix-name == null
-              then ""
-              else "${compiler-nix-name}:";
-          fixFlakeOutput = name: output:
-            if name == "devShell"
-            then output
-            else lib.mapAttrs' (n: v: lib.nameValuePair (prefix + n) v) output;
-          fixFlake =
-            lib.mapAttrs fixFlakeOutput;
-        in
-          fixFlake (project.flake {});
-
-      combineOutputs = name:
-        if name == "devShell"
-        then lib.last
-        else lib.foldl (l: r: l // r) {};
-
-    in lib.zipAttrsWith combineOutputs (map flakeForGhc (ghcVersions ++ [null]));
-
+  makeCI = haskellPkgs: {
+    # haskell project root
+    src,
+    # haskell package names
+    packageNames,
```

Comment:
Can you check if this can be deduced automatically from the result of `cabalProject` or `stackProject` call to reduce the necessity to list packages manually? Asking because previously I once forgot to add the package name to such list and as a result this package wasn't tested by the CI :sweat_smile:

Reply:
We can use the flake' attribute, it provides everything we need except haddock build, but haddock build should be covered by the `doHaddock` module option, which defaults to `true`